addons/analytic/models/analytic_plan.py

```diff
@@ -266,30 +303,55 @@ def _sync_all_plan_column(self):
     def _sync_plan_column(self, model):
         # Create/delete a new field/column on related models for this plan, and keep the name in sync.
         for plan in self:
-            prev = plan._find_plan_column(model)
-            if plan.parent_id and prev:
-                prev.unlink()
-            elif prev:
-                prev.field_description = plan.name
-            elif not plan.parent_id:
-                column = plan._strict_column_name()
-                field = self.env['ir.model.fields'].with_context(update_custom_fields=True).sudo().create({
-                    'name': column,
-                    'field_description': plan.name,
-                    'state': 'manual',
-                    'model': model,
-                    'model_id': self.env['ir.model']._get_id(model),
-                    'ttype': 'many2one',
-                    'relation': 'account.analytic.account',
-                    'copied': True,
-                    'on_delete': 'restrict',
-                })
-                Model = self.env[model]
-                if Model._auto:
-                    tablename = Model._table
-                    indexname = make_index_name(tablename, column)
-                    create_index(self.env.cr, indexname, tablename, [column], 'btree', f'{column} IS NOT NULL')
-                    field['index'] = True
+            prev_stored = plan._find_plan_column(model)
+            depth, name_related = plan._hierarchy_name()
+            prev_related = plan._find_related_field(model)
+            if plan.parent_id:
+                if prev_stored:
```

Comment:
You should definitely add a comment here or in this method in general, explaining what (and maybe why) we're doing it.
In the context of this PR, I understand. I'm not sure I would again in 1 year.
Especially, why it's different in the case of the parent or not

Maybe too on `_subplan_used`. Other methods seemed "clear" (:D) from the get go

Reply:
Do these comments help?

Reply:
Yes, I think.
Thanks